trpcultivate_phenotypes/tests/src/Functional/ConfigRRulesTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @file
+ * Functional test of R Transfomration rules configuration page.
+ */
+
+namespace Drupal\Tests\trpcultivate_phenotypes\Functional;
+
+use Drupal\Component\Utility\SafeMarkup;
+use Drupal\Tests\tripal_chado\Functional\ChadoTestBrowserBase;
+
+ /**
+  *  Class definition ConfigRRulesTest.
+  */
+class ConfigRRulesTest extends ChadoTestBrowserBase {
+  const SETTINGS = 'trpcultivate_phenotypes.settings';
+
+  /**
+   * Modules to enabled
+   * 
+   * @var array
+   */
+  protected static $modules = ['trpcultivate_phenotypes'];
+
+  /**
+   * Admin user with admin privileges.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $admin_user;
+
+  /**
+   * Test R transformation rules configuration page.
+   */
+  public function testForm() {
```

Comment:
This is a nice functional review :-)

⚠️ 
You'll also want to add a call to specific methods in the controllers as more of a unit test. The functional test confirms from a user perspective and the unit test confirms from a developer API perspective. Also, you'll notice your code coverage is more focused on the unit test versus the functional one... so this PR has very low test coverage despite the good functional test.

Reply:
Prepared functional test and unit test to RRules configuration